src/dodal/devices/eiger.py

```diff
@@ -190,29 +201,46 @@ def set_num_triggers_and_captures(self) -> Status:
         )
         return status
 
-    def wait_for_stale_parameters(self):
-        await_value(self.stale_params, 0).wait(self.STALE_PARAMS_TIMEOUT)
+    def wait_for_stale_parameters(self, statuses):
+        # await_value(self.stale_params, 0).wait(self.STALE_PARAMS_TIMEOUT)
+        statuses[0] = await_value(self.stale_params, 0)
+
+    def wait_for_odin_status(self, statuses):
+        statuses[1] = self.odin.file_writer.capture.set(1)
+        statuses[1] &= await_value(self.odin.meta.ready, 1)
+
+    def wait_for_cam_acquire(self, statuses):
+        LOGGER.info("Setting aquire")
+        statuses[2] = self.cam.acquire.set(1)
+
+    def await_value(self, statuses):
+        statuses[3] = await_value(self.odin.fan.ready, 1)  # timeout 10?
+        pass
 
     def forward_bit_depth_to_filewriter(self):
         bit_depth = self.bit_depth.get()
         self.odin.file_writer.data_type.put(f"UInt{bit_depth}")
 
-    def arm_detector(self):
+    def arm_detector(self) -> Status:
+        statuses = [Status(1), Status(2), Status(3), Status(4)]
```

Comment:
Should: Each function that we pass these to is overwriting the entry so it's a bit confusing having these here. Might be better to start with an empty list then each function can just append to the list.